### PR TITLE
Updated the rain gauge and moved temp/hum/temp+hum from Weatherwidget to TemperatureWidget

### PR DESCRIPTION
**Pull request by gerard33**
**Rain gauge**
Updated the rain gauge. It rained so I could work on it :-)
- show correct values from MQTT
- show 1 decimal
- changed icon

rain gauge-new

**Temperature**
Moved temp, hum and temp + hum from WeatherWidget to TemperatureWidget.
This uses the GenericWidget and adds icons.
Barometer is still in WeatherWidget due to its own icons.

temperature widget

### CODE REVIEW

**Comment by gerard33** [comment]
This rain gauge in the RainWidget is now showing the correct values.

I noticed a strange thing however.
My virtual rain gauge from Weather Under shows a "normal" value for the rain, 2.0mm in the example in Domoticz and also 2.0 in the MQTT svalue2.
The MQTT stype is WWW.

My rain gauge hardware device (https://www.weerstationkopen.nl/accessoires-weerstations/homewizard/homewizard-regenmeter-ts906) however shows 2.8mm in Domoticz, but 1528.8mm in the MQTT svalue2.
The value in Domoticz is the the new value (1528.8) minues the previous value (1526).
The MQTT stype is TFA.

I think the only way to solve this is by changing the MQTT message.

**Comment by gerard33** [comment]
The rain icon (opacity) is used for the rain gauge.

**Comment by t0mg** [comment]
Thanks, nice work!

Regarding TFA stype, I think you're right, unless this 1526 value is a constant (like a calibration offset) that we could get from the JSON once and for all. If it really is just "previous" then I think all we can do is requesting a change in Domoticz.

File: src/widgets/TemperatureWidget.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          value2={this.props.hum !== undefined && <div className="hum">{this.props.hum}</div>}
+          {...this.props} />
+        );
+      case "Humidity" :
+        const humidityStatus = ((this.props.humstat === "0") ? 'Normal' : (this.props.humstat === "1") ? 'Comfortable' : (this.props.humstat === "2") ? 'Dry' : 'Wet');
+        return (
+        <GenericWidget class="TemperatureWidget" icon="humidity"
+          isOn={this.props.humstat > 2}
+          value1={<div className="humonly">{this.props.humonly}</div>}
+          value2={humidityStatus}
```
Comment (t0mg) [inline src/widgets/TemperatureWidget.js]: I think you don't need to access the dtype again here and do another switch. You can simply test this.props.humonly in an if-else statement.

Comment (gerard33) [reply src/widgets/TemperatureWidget.js]: Can you explain this a little bit more please? I don't exactly understand what you mean.

Comment (t0mg) [reply src/widgets/TemperatureWidget.js]: You could replace the switch (device.dtype) logic by a simple if else statement. 
Before:

```
switch (device.dtype) {
  case "Temp":
  case "Temp + Humidity":
    return widget 1
  case "Humidity":
    return widget 2
}
```
After:
```
if (this.props.humonly) {
  return widget 2
}
return widget 1
```

Comment (gerard33) [reply src/widgets/TemperatureWidget.js]: That's indeed a nice solution! Will update the code for that part.

**Comment by gerard33** [comment]
Just saw that the humidity only device with value 0% is not working as it should (it now shows as a temp icon). Let me have a look at this so I can update the code before you accept this pull request.

Edit: added a !== undefined and now it is working. The pull request can be accepted.

**Comment by t0mg** [comment]
Ok thanks for updating this, nice work!

**Comment by t0mg** [comment]
Oh by the way, in the future please use the original name of the icons from material.io, this makes it easier to figure out what they look like when reading the code, and whether an icon is already used when adding new ones :)

**Comment by gerard33** [comment]
Will do :)

**Comment by Sneeuwtijger** [comment]
A suggestion - the temperature icon is now a Sun - that's how I interpret it ;) is it an idea to change it to a thermometer like icon - something like in Domoticz default theme? Now it is confusing, a bit, to see the cloud icon for the Weather Widget next to the sun icon of the Temperature Widget.
Example - https://materialdesignicons.com/api/download/icon/svg/A9FC898C-8CC5-44F2-822D-05C5625E95C8

**Comment by t0mg** [comment]
Nice find @Sneeuwtijger! The license seems OK and they have extra stuff on top of the Google icons, like this thermometer icon. Given that it's also provided as npm package, I might consider swicthing over to just using this with the provided css. It would make Reacticz build much larger though (the eof font file is 262Kb, the svg file is 2Mb). I'll give it some thought.

**Pull request merged**